Port Troubleshooting Section from Screensteps (#131)
wpilibsuite/frc-docs: source/docs/software/driverstation/programming-radios-for-fms-offseason.rst

Programming Radios for FMS Offseason
====================================

When using the FMS Offseason software, the typical networking setup is to use a single access point with a single SSID and WPA key. This means that the radios should all be programmed to connect to this network, but with different IPs for each team. The Team version of the FRC Bridge Configuration Utility has an FMS-Lite mode that can be used to do this configuration.

Before you begin using the software:

#. Disable WiFi connections on your computer, as it may prevent the configuration utility from properly communicating with the bridge
#. Make sure no devices are connected to your computer via ethernet, other than the wireless bridge.

Pre-Requisites
--------------

> 

Reviewer: ```suggestion

.. note:: While many other WPILib software uses Java 11, the FRC Radio Configuration Utility requires Java 8. 
```

> The 2016 FRC Radio Configuration Utility requires the Java Runtime Engine (JRE). If you do not have Java installed, you can download the JRE from here: https://www.java.com/en/download/

Reviewer: ```suggestion
The FRC Radio Configuration Utility requires the Java Runtime Engine (JRE). If you do not have Java installed, you can download the JRE from `here <https://www.java.com/en/download/>`__.
```

The FRC Radio Configuration Utility requires Administrator privileges to configure the network settings on your machine. The program should request the necessary privileges automatically (may require a password if run from a non-Administrator account), but if you are having trouble try running it from an Administrator account.

Application Notes
-----------------

> The 2016 Radio Kiosk will program the radio to enforce the 7 Mbps bandwidth limit on traffic exiting the radio over the wireless interface. In the home configuration (AP mode) this is a total, not a per client limit. This means that streaming video to multiple clients is not recommended.

Reviewer: ```suggestion
The Radio Kiosk will program the radio to enforce the 4 Mbps bandwidth limit on traffic exiting the radio over the wireless interface. In the home configuration (AP mode) this is a total, not a per client limit. This means that streaming video to multiple clients is not recommended.
```

> The 2016 Kiosk has been tested on Windows 7, 8 and 10. It may work on other operating systems, but has not been tested.

Reviewer: ```suggestion
The Kiosk has been tested on Windows 7, 8 and 10. It may work on other operating systems, but has not been tested.
```

Programmed Configuration
^^^^^^^^^^^^^^^^^^^^^^^^

.. image:: images/programming-radios-for-fms-offseason-1.png

The Radio Configuration Utility programs a number of configuration settings into the radio when run. These settings apply to the radio in all modes (including at events). These include:

- Set a static IP of 10.TE.AM.1
- Set an alternate IP on the wired side of 192.168.1.1 for future programming
- Bridge the wired ports so they may be used interchangeably
- The LED configuration noted in the graphic above
- 4Mb/s bandwidth limit on the outbound side of the wireless interface
- QoS rules for internal packet prioritization (affects internal buffer and which packets to discard if bandwidth limit is reached). These rules are Robot Control and Status (UDP 1110, 1115, 1150) >> Robot TCP & Network Tables (TCP 1735, 1740) >> Bulk (All other traffic).

When programmed with the team version of the Radio Configuration - Utility, the user accounts will be left at (or set to) the firmware - defaults **for the DAPs only**:

- Username: root
- Password: root

> Note: It is not recommended to modify the configuration manually

Reviewer: ```suggestion
.. note:: It is not recommended to modify the configuration manually
```

Download the software
---------------------

.. image:: images/programming-radios-for-fms-offseason-2.png

Download the latest FRC Radio Configuration Utility Installer from the `WPILib project File Releases <https://usfirst.collab.net/sf/frs/do/listReleases/projects.wpilib/frs.frc_radio_configuration_utility>`__.

Install the software
--------------------

.. image:: images/programming-radios-for-fms-offseason-3.png

Double click on FRC_Radio_Configuration_MM_DD_YY.exe to launch the installer. Follow the prompts to complete the installation.

Part of the installation prompts will include installing WinPCap if it is not already present. The WinPCap installer contains a checkbox (checked by default) to start the WinPCap driver on boot. You should leave this box checked.

Launch the software
-------------------

.. image:: images/programming-radios-for-fms-offseason-4.png

Use the Start menu or desktop shortcut to launch the program.

> .. note:: If you need to locate the program it is installed to C:/Program Files (x86)/FRC Radio Configuration Utility. For 32-bit machines the path is C:/Program Files/FRC Radio Configuration Utility/

Reviewer: ```suggestion
.. note:: If you need to locate the program it is installed to ``C:/Program Files (x86)/FRC Radio Configuration Utility``. For 32-bit machines the path is ``C:/Program Files/FRC Radio Configuration Utility/``
```

Programming Radios for FMS Offseason
====================================

When using the FMS Offseason software, the typical networking setup is to use a single access point with a single SSID and WPA key. This means that the radios should all be programmed to connect to this network, but with different IPs for each team. The Team version of the FRC Bridge Configuration Utility has an FMS-Lite mode that can be used to do this configuration.

Before you begin using the software:

#. Disable WiFi connections on your computer, as it may prevent the configuration utility from properly communicating with the bridge
#. Make sure no devices are connected to your computer via ethernet, other than the wireless bridge.

Pre-Requisites
--------------

> .. note:: While many other WPILib software uses Java 11, the FRC Radio Configuration Utility requires Java 8. 

Reviewer: ```suggestion
.. note:: While many other WPILib software uses Java 11, the FRC Radio Configuration Utility requires Java 8.

```
